markdown_lab/network/client.py

```
@@ -0,0 +1,293 @@
+from requests.adapters import HTTPAdapter
+
+from ..core.config import MarkdownLabConfig
+from ..core.errors import NetworkError, handle_request_exception
+from ..core.throttle import RequestThrottler
+
+logger = logging.getLogger(__name__)
+
+
+class HttpClient:
+    """Unified HTTP client with retry logic, rate limiting, and error handling.
+    
```

Ruff (0.11.9)

Blank line contains whitespace

Remove whitespace from blank line

(W293)